### ISSUE TITLE
to think about - state-colors

### ISSUE THREAD

**Issue by vogelor** [tasteui/tasteui#154]
this is NOT a issue, only something to think about.

Maybe it is worth to think about semantic colors.
i mean
color-danger
color-warning
color-info
color-success
and so on.

this is to be sure, that every error (input, alert, modal,...) has the same color.

like here:
https://daisyui.com/docs/colors/

so i think it is easier if there are more and more contributors for this great UI

**Comment by devajmeireles** [comment]
@vogelor , thanks for all your suggestions. Please, continue with these incredible contributions.

After #153 is merged, you'll be able to create custom colors by using the deep personalization concept. Something like this:

`config/tallstackui.php`:

```php
    // ...
    'components' => [
       // 'alert' => \TallStackUi\View\Components\Alert::class,
       'alert' => \App\View\Components\CustomAlert::class,
```

`\App\View\Components\CustomAlert`:

```php
class CustomAlert extends \TallStackUi\View\Components\Alert
{
    public function backgroundColor() : array 
    {
        return [
            'solid' => [
                'success' => 'bg-green-300 ...'
            ],
        ];
    }

    public function textColor() : array 
    {
        return [
            'solid' => [
                'success' => 'text-green-600 ...'
            ],
        ];
    }
}
```

Using:

```blade
<x-alert ... color="success" />
```

Is this what you expected with this suggestion?

**Comment by vogelor** [comment]
yes - but i think it is better if YOU add this to the core!
adding this to the core means if any contributor wants to write a new component (for example a progressbar ;-) she can use the semantic color (error, info, success) without thinking about what all the other are using for tailwind classes...

is the error in Input bg-red-500 - or bg-darkred-300 or bg-lightred-400 (these are pseudo colors) ??? don't know at all. BUT i know it is bg-error

if we all have the same semantic colors to work with (error, warning, info, success) we all have it easier to code new components. because bg-info or text-info is ALWAYS the same but bg-green-500 differs from bg-green-300 

it is about consistency if we all use semantic colors (error, warning, info, success) instead of red, blue, fuchsia, pink, rose,...

**Comment by vogelor** [comment]
so one thing is what the "end programmer" can do (Custom-colors) and the other is what we all contributors can use to have always the same color without thinking about it

And the next benefit is:
if we say
color.error = color.red;
the "End-Programmer" can easily change it to
color.error = color.rose;

to change ALL error - colors globaly to better customize it...

**Comment by vogelor** [comment]
and (to think more to the future) if you have this colors
error
warning
info
success
primary
secondary
tertiary
dark

then it is easy for us to create themes - how cool is that...

**Comment by vogelor** [comment]
i am very Sorry - i have overseen something in your example.

this is the right code...

```php
class CustomAlert extends \TallStackUi\View\Components\Alert
{
    public function backgroundColor() : array 
    {
        return [
            'solid' => [
                'success' => 'bg-success-300 ...'
            ],
        ];
    }

    public function textColor() : array 
    {
        return [
            'solid' => [
                'success' => 'text-success-600 ...'
            ],
        ];
    }
}

```
so not bg-green-300 but bg-success-300
because YOU define color.success = color.green in the tailwind.config.js!

**Comment by devajmeireles** [comment]
> i am very Sorry - i have overseen something in your example.
> 
> this is the right code...
> 
> ```
> class CustomAlert extends \TallStackUi\View\Components\Alert
> {
>     public function backgroundColor() : array 
>     {
>         return [
>             'solid' => [
>                 'success' => 'bg-success-300 ...'
>             ],
>         ];
>     }
> 
>     public function textColor() : array 
>     {
>         return [
>             'solid' => [
>                 'success' => 'text-success-600 ...'
>             ],
>         ];
>     }
> }
> ```
> 
> so not bg-green-300 but bg-success-300 because YOU define color.success = color.green in the tailwind.config.js!

Exactly! You just need to create the `success` in your tailwind configuration, and then apply it like this example.

Due to our policy, I'll close this issue. Wait for the next big release, which contains all our works of #153 , and then you will be able to do that. Feel free to open a new issue whenever necessary.

**Comment by vogelor** [comment]
i am afraid my English is to bad so that you don't understand what i mean.
yes i can do this. this is not the problem. but if I do this and not YOU, than this project is unable to take the benefit of semantics colors (in addition to the named colors). With semantics colors it is very easy to be sure, that every component has the same tailwind-color for error. No matter if it is a checkbox, a switch, a input or something else. think about a new component for example a link. and then thing about a link, displayed when a error occurred. i a new programmer wants to code this link she didn't know, what color YOU used, when you coded the input-component. so the programmer has to take a look at your code of the input and to look, what tailwind-color (class) YOU are using for the input error-state. He then has to copy this class inside hist own link class.
wouldn`t it be much easier to know that the color has the tailwind-class "text-error" and "bg-error".
i mean, YOU have a tailwind.config.js inside YOUR project, why not defining this semantics color inside YOUR tailwind config and then using this tailwind classes inside TALLSTACKUI components.
i believe then all developers helping you have a easier coding KNOWING the color and not guessing what you had done.

if you decide NOT to do it then this is O.K. it is your project.
But i want to be sure that you understand what i mean and then say no - this is O.K.
i dont like the idea that you decide no without understanding what i mean...
sorry again - it is YOUR Project...
